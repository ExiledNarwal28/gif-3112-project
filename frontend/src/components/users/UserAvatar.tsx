import React from 'react';
import { Avatar } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  defaultSize: {
    width: theme.spacing(9),
    height: theme.spacing(9),
  },
}));

interface userAvatarProps {
  src?: string;
  size?: string;
  userName: string;
}

const UserAvatar = (props: userAvatarProps) => {
  const classes = useStyles();
  const { src, userName, size } = props;

  if (src) {
    return (
      <div>
        <Avatar className={size || classes.defaultSize} src={src} />
      </div>
    );
  }

  return (
    <div>
      <Avatar className={size || classes.defaultSize}>
        {userName.charAt(0)}
      </Avatar>
    </div>
  );
};

UserAvatar.defaultProps = {
  src: null,
  size: null,
};

export default UserAvatar;
