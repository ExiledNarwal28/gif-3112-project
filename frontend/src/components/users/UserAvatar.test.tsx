import React from 'react';
import { render } from '@testing-library/react';
import { Avatar } from '@material-ui/core';
import UserAvatar from './UserAvatar';

test('Avatar contains image when src prop is passed', () => {
  const userAvatar = renderer.create(
    <UserAvatar
      src="https://interactive-examples.mdn.mozilla.net/media/cc0-images/grapefruit-slice-332-332.jpg"
      userName="Omar"
    />
  );

  const tree = userAvatar.toJSON();

  expect(tree).toMatchSnapshot();
});

test('Avatar contains username first letter when src prop is absent', () => {
  const userAvatar = renderer.create(<UserAvatar userName="Test" />);

  const tree = userAvatar.toJSON();

  expect(tree).toMatchSnapshot();
});

describe('When rendering UserAvatar', () => {
  it('Should render Avatar', () => {
    render(<Avatar />);
  });
});
