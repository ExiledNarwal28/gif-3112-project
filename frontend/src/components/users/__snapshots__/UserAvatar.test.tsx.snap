// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar contains image when src prop is passed 1`] = `
<div>
  <div
    className="MuiAvatar-root MuiAvatar-circle makeStyles-defaultSize-1"
  >
    <img
      className="MuiAvatar-img"
      src="https://interactive-examples.mdn.mozilla.net/media/cc0-images/grapefruit-slice-332-332.jpg"
    />
  </div>
</div>
`;

exports[`Avatar contains username first letter when src prop is absent 1`] = `
<div>
  <div
    className="MuiAvatar-root MuiAvatar-circle makeStyles-defaultSize-1 MuiAvatar-colorDefault"
  >
    T
  </div>
</div>
`;
